src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/BlazorDebug/BlazorDebugConfigurationProvider.ts

```diff
@@ -0,0 +1,94 @@
+
+    public async resolveDebugConfiguration(folder: vscode.WorkspaceFolder | undefined, configuration: vscode.DebugConfiguration): Promise<vscode.DebugConfiguration | undefined> {
+        const shellPath = process.platform === 'win32' ? 'cmd.exe' : 'dotnet';
+        const shellArgs = process.platform === 'win32' ? ['/c', 'chcp 65001 >NUL & dotnet run'] : ['run'];
+        const spawnOptions = {
+            cwd: configuration.cwd || folder && folder.uri && folder.uri.fsPath,
+            env: {
+                ...process.env,
+                ASPNETCORE_ENVIRONMENT: 'Development',
+                ...configuration.env,
+            },
+        };
+
+        const output = this.vscodeType.window.createTerminal({
+          name: 'Blazor WebAssembly App',
+          shellPath,
+          shellArgs,
+          ...spawnOptions,
+        });
+
+        /**
+         * On non-Windows platforms, we need to terminate the Blazor
+         * dev server and its child processes.
+         */
+        const terminate = this.vscodeType.debug.onDidTerminateDebugSession(async event => {
+            await onDidTerminateDebugSession(event, this.logger, await output.processId);
+            terminate.dispose();
+        });
+
+        output.show();
+
+        const browser = {
+            name: '.NET Core Debug Blazor Web Assembly in Browser',
+            type: configuration.browser === 'edge' ? 'pwa-msedge' : 'pwa-chrome',
+            request: 'launch',
+            timeout: configuration.timeout || 30000,
+            url: configuration.url || 'https://localhost:5001',
+            webRoot: configuration.webRoot || '${workspaceFolder}',
+            inspectUri: '{wsProtocol}://{url.hostname}:{url.port}/_framework/debug/ws-proxy?browser={browserInspectUri}',
+            trace: configuration.trace || false,
+            noDebug: configuration.noDebug || false,
+        };
+
+        let showErrorInfo = false;
+
+        try {
+            /**
+             * The browser debugger will immediately launch after the
+             * application process is started. It waits a `timeout`
+             * interval before crashing after being unable to find the launched
+             * process.
+             *
+             * We do this to provide immediate visual feedback to the user
+             * that their debugger session has started.
+             */
+            await this.vscodeType.debug.startDebugging(folder, browser);
+            this.logger.logVerbose('[DEBUGGER] Launching browser debugger...');
+        } catch (error) {
+            this.logger.logError(
+                '[DEBUGGER] Error when launching browser debugger: ',
+                error,
+            );
+            showErrorInfo = true;
+        }
+
+        if (showErrorInfo) {
+            const message = `There was an unexpected error while launching your debugging session. Check the console for helpful logs and visit https://aka.ms/blazorwasmcodedebug for more info.`;

```

Comment:
Is the link clickable in the pop-up? If not it might be better to provide two buttons in the error message pop-up (Ok and Learn More). For example: https://github.com/OmniSharp/omnisharp-vscode/blob/74cb376bc96ed9959f1fd0ca1dba5c86caa3c9ba/src/omnisharp/utils.ts#L89-L95